### PR TITLE
fix: keyboard navigation in search bar component

### LINKED ISSUE #13039: Keyboard Navigation Bug: Multiple Options Highlighted, Always Selects First
**Describe the bug**
When we use the keyboard to move up and down in the search dropdown, 
sometimes more than one option looks highlighted at the same time.

 Also, when we press Enter, it always picks the first option and get highlighted (green color) , instead of selected one.

**To Reproduce**

1. Go to 'Search Patients' page.

https://github.com/user-attachments/assets/89a41382-bd5a-4a80-b769-3597c2135da9

**Expected behavior**
Only selected option shoud get highlight instead of first option and removing the shading for multiple options.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

---

### Instructions for Requesting Assignment:

To request assignment, please clearly outline your solution and timeline by commenting on the issue using the format below:

**Describe your solution clearly:**
Provide a detailed explanation of your proposed solution, including your approach, key implementation steps, and relevant examples or references. Mention any dependencies, assumptions, or risks you foresee that might affect your timeline or implementation.

**Expected Timeline:**
- End date: [Expected submission date of a completed Pull Request]

**Additional Context:**
Include any other relevant context, links, screenshots, or resources that support your proposed solution.

> 🚨 Your assignment may be unassigned if there is no activity or progress within the stated timeline unless communicated clearly and agreed upon.

### PR DESCRIPTION
**Pull request by shiv7shukla**
## Proposed Changes

Fixes #13039
- Switched to simpler CommandItem behavior for focus/selection
- Enter selection is handled by CommandItem’s built-in behavior, triggered when the item is clicked or selected
- The old version maintained focusedIndex and updated it on ArrowUp, ArrowDown, and Enter. The new version removes this state entirely, eliminating custom index-tracking logic.

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is placed in I18n files.
- [ x ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ x ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes
 

https://github.com/user-attachments/assets/e20e3205-36ea-4b4a-b01c-f4af5fe69670

## Summary by CodeRabbit

* **Refactor**
  * Simplified search input navigation: removed per-item focus tracking so hover highlights items, a persistent Enter cue appears for the selected item, and choosing an option closes the popover and returns focus to the input. Added a test-friendly identifier on the primary option button.

* **Tests**
  * Added end-to-end keyboard navigation tests for patient search to verify arrow-key highlighting, Home/End behavior, Enter to commit selection, boundary handling, empty results, and selection persistence after reopening.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by parvathyns-creator** [comment]
@nihal467 This looks fine

**Review by SiddanthNayak — APPROVED** [review]
LGTM

**Review by nihal467 — CHANGES_REQUESTED** [review]
- functionality looks good, add a playwright test to verify the behavior as well

**Comment by builtby-SHIV** [comment]
> * functionality looks good, add a playwright test to verify the behavior as well

@nihal467 I've added the playwright test please review. Also can you please provide the facility ID that I should hardcode in the test since its an issue raised by coderabbit

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    await page.keyboard.press("ArrowDown");
+    const highlightedText = await commandItems.nth(1).innerText();
+    await expect(commandItems.nth(1)).toHaveAttribute("data-selected", "true");
+    await page.keyboard.press("Enter");
+    await expect(commandItems).toHaveCount(0);
+    await page.keyboard.press("Control+K");
+    const selectedButton = page.locator("button:has(svg)");
+    await expect(selectedButton).toContainText(highlightedText);
+  });
+});
```
[the remark this answers is not in the document]
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: Only keyboard navigation while highlighting an option and selecting it is supposed to be tested

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { expect, test } from "@playwright/test";
+
+test.describe("Keyboard navigation in search patients", () => {
+  test("keyboard navigation: arrow keys highlight, enter commits selection", async ({
+    page,
+  }) => {
+    await page.goto("/facility/:facilityId/patients");
+    await page.keyboard.press("Control+k");
```
[the remark this answers is not in the document]
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: uppercase k is opening the search bar in browser

[the remark this answers is not in the document]
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: uppercase k is opening the browser search bar which might also be the case in other browsers so I'm going with lowercase k

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { expect, test } from "@playwright/test";
+
+test.describe("Keyboard navigation in search patients", () => {
+  test("keyboard navigation: arrow keys highlight, enter commits selection", async ({
+    page,
+  }) => {
+    await page.goto("/facility/:facilityId/patients");
```
Comment (SiddanthNayak) [inline tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]:
```suggestion
    const facilityId = getFacilityId();
    await page.goto(`/facility/${facilityId}/patients`);
```

@shiv7shukla look at other test files how facilityId is retrieved and used.

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { expect, test } from "@playwright/test";
```
Comment (SiddanthNayak) [inline tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: The test is for facility right? 

So the test file should be in `tests/facility/patient` ...

**Review by SiddanthNayak — CHANGES_REQUESTED** [review]

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts (hunk 4 above)
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: @SiddanthNayak It's for the patients' section where the particular navigation bug is arriving while searching for the patients

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts (hunk 3 above)
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: @SiddanthNayak I've done the changes please review it

**Comment by builtby-SHIV** [comment]
@SiddanthNayak @nihal467 please review and suggest if any changes are required

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SiddanthNayak) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: @shiv7shukla why was this resolved? 

Can you address comment 1). 

All setup such as getFacilityId and navigation must be done in `beforeEach`. 

Refer other test files to see how it is done..

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { getFacilityId } from "tests/support/facilityId";
+
+const facilityId = getFacilityId();
+
+test.describe("Keyboard navigation in search patients", () => {
+  test("keyboard navigation: arrow keys highlight, enter commits selection", async ({
+    page,
+  }) => {
+    await page.goto(`/facility/${facilityId}/patients`);
```
Comment (SiddanthNayak) [inline tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: add `beforeEach` function. 

getFacilityId and navigation must be done in `beforeEach`. 

Refer other test files to see how it is done.

**Review by SiddanthNayak — CHANGES_REQUESTED** [review]

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: @SiddanthNayak I've added multiple navigation scenarios and edge cases as suggested by coderabbit but can you help me in writing the test data setup. It's my first time writing a test.

File: tests/organization/patient/searchpatient/keyboardNavigation.spec.ts (hunk 5 above)
Comment (shiv7shukla) [reply tests/organization/patient/searchpatient/keyboardNavigation.spec.ts]: @SiddanthNayak Done please review it

**Comment by bodhish** [comment]
@shivankacker & @SiddanthNayak review?

**Review by SiddanthNayak — APPROVED** [review]
LGTM!

With regards to the spec file, the tests are being written for Patient search on Facility Level and not organization level. 

So I think the test file must be in `tests/facility/patient` instead.

cc: @nihal467

**Comment by builtby-SHIV** [comment]
@nihal467 @SiddanthNayak While trying to resolve the merge conflicts by mistake I closed this PR. The new PR is at: https://github.com/ohcnetwork/care_fe/pull/16079
